core/test-runtime/src/genesismap.rs

```diff
@@ -29,6 +29,10 @@ pub struct GenesisConfig {
 	pub authorities: Vec<AuthorityId>,
 	pub balances: Vec<(AccountId, u64)>,
 	pub heap_pages_override: Option<u64>,
+	/// Additional storage key pairs that will be added to the genesis map.
+	pub extra_storage: Vec<(Vec<u8>, Vec<u8>)>,
+	/// Prevent direct construction of this object.
+	_use_ctor: std::marker::PhantomData<()>,
```

Comment:
Why do we need this? Can't we just make the fields private or crate-public?
If not then I think `_use_ctor:()` would work equally well :)

Reply:
I think crate-public won't get the effect I wanted. But I didn't even try to use private fields because I didn't want to wait for a compile cycle. Regarding `PhantomData` and `()` yeah that was just an unconscious brainfart.

Reply:
yep, it seems that there were no reason to keep them in non-private!